File: Intermediate/TemplaterJson/src/SupportedType.cs
using System.Collections.Generic;
using System.IO;

namespace TemplaterJson
{
	internal static class SupportedType
	{
		private readonly static Dictionary<string, string> Extensions = new Dictionary<string, string>();
		static SupportedType()
		{
			Extensions["docm"] = "Macro-Enabled Office Open XML document (Word)";
			Extensions["docx"] = "Office Open XML document (Word)";
			Extensions["xlsm"] = "Macro-Enabled Office Open XML spreadsheet (Excel)";
			Extensions["xlsx"] = "Office Open XML spreadsheet (Excel)";
			Extensions["csv"] = "Comma separated values document";
			Extensions["txt"] = "Text file (native encoding)";
			Extensions["utf8"] = "Text file (UTF-8 encoding)";
		}

		public static string FindExtension(string file)
		{
			var ind = file.LastIndexOf('.');
			if (ind == -1) return null;
			var ext = file.Substring(ind + 1).ToLowerInvariant();
			if (Extensions.ContainsKey(ext)) return ext;
			return null;
		}

		public static void ShowHelp(TextWriter writer)
		{
			writer.WriteLine("Supported extensions:");
			foreach (var kv in Extensions)
				writer.WriteLine(kv.Key + " - " + kv.Value);
		}
	}
}

File: Intermediate/TemplaterJson/src/main/java/hr/ngs/templater/SupportedType.java
package hr.ngs.templater;

import java.util.Locale;

enum SupportedType {
    DOCM("Macro-Enabled Office Open XML document (Word)"),
    DOCX("Office Open XML document (Word)"),
    XLSM("Macro-Enabled Office Open XML spreadsheet (Excel)"),
    XLSX("Office Open XML spreadsheet (Excel)"),
    PPTM("Macro-Enabled Office Open XML presentation (PowerPoint)"),
    PPTX("Office Open XML presentation (PowerPoint)"),
    CSV("Comma separated values document"),
    TXT("Text file (native encoding)"),
    UTF8("Text file (UTF-8 encoding)");

    public final String description;
    public final String extension;

    SupportedType(final String description) {
        this.description = description;
        this.extension = name().toLowerCase(Locale.ENGLISH);
    }

    public static SupportedType getByFilename(final String path) {
        final int lastIndexOfDot = path.lastIndexOf('.');
        if (lastIndexOfDot == -1) return null;

        final String extension = path.substring(lastIndexOfDot + 1);
        return valueOf(extension.toUpperCase(Locale.ENGLISH));
    }
}
